crates/cargo-lambda-watch/src/lib.rs

```diff
@@ -94,9 +94,18 @@ pub async fn run(
         None
     };
 
-    let binary_packages =
+    let mut binary_packages =
         filter_binary_targets_from_metadata(metadata, kind_bin_filter, package_filter);
 
+    if let Some(func) = config.cargo_opts.bin.first() {
```

Comment:
This might not be the best way to handle it because the `--bin` flag can be specified multiple times. This change would completely ignore the rest of the values.

It's probably better to create a filter like the code above does with the `cargo_options.packages` for `cargo_options.bin`. There is logic later that detects if there is only one element in `binary_packages` to prevent adding the prefix to the url.

Reply:
Wasn't aware of that behaviour in Cargo. Yeah, a filter makes sense, will take a look.

Reply:
Something like this might work:

```rust
    let bins = cargo_options.bin.clone();
    let binary_filter =
        move |t: &CargoTarget| kind_bin_filter(t) && (bins.is_empty() || bins.contains(&t.name));

    let binary_packages =
        filter_binary_targets_from_metadata(metadata, binary_filter, package_filter);
```

Reply:
Thanks for the suggestion. Updated accordingly.

I'll start updating the docs. Is there anything other than the docs? Can this be tested? Do I need to bump the version number manually or is that automated?

Reply:
I'm planning a release in #764   so you don't have to worry about package version number. It could be good to extract all that into an isolated function for testing purposes.